arch/risc-v/src/qemu-rv/qemu_rv_head.S

```diff
@@ -77,11 +77,20 @@ __start:
   ld   sp, 0(t0)
 #endif
 
-  /* sp (stack top) = sp + idle stack size */
-
-  li   t0, CONFIG_IDLETHREAD_STACKSIZE
+  /*
+   * sp (stack top) = sp + idle stack size - XCPTCONTEXT_SIZE
+   * 
+   * Note: Reserve some space used by up_initial_state since we are already
+   * running and using the per CPU idle stack.
+   */
+
+  li   t0, CONFIG_IDLETHREAD_STACKSIZE - XCPTCONTEXT_SIZE
```

Comment:
change to STACK_ALIGN_UP(CONFIG_IDLETHREAD_STACKSIZE - XCPTCONTEXT_SIZE)
and remove line 90-93